bindings/c/library.cpp

```diff
@@ -490,7 +493,7 @@ TORRENT_EXPORT int torrent_get_status(int tor, torrent_status* s, int struct_siz
 	s->state = (state_t)ts.state;
 	s->paused = ts.paused;
 	s->progress = ts.progress;
-	strncpy(s->error, ts.error.c_str(), 1025);
+	strncpy(s->error, ts.error.c_str(), 1024);
```

Comment:
```suggestion
	strncpy(s->error, ts.error.c_str(), sizeof(s->error)-1);
	s->error[sizeof(s->error)-1] = '\0';
```

Reply:
I wonder if the original intention was to say 1023 instead of 1025, and assume the array was initialized to zeroes. Either way, I think you're right that it needs to be explicitly null terminated.

Reply:
Dangerous assumption)
Perhaps we could try to re-analyze some other places in code to fix potential memory overrun and integrate it in this change? Any suspects?